wire-ios-sync-engine/Source/UserSession/ZMUserSession/ZMUserSession.swift

```diff
@@ -552,27 +554,22 @@ public final class ZMUserSession: NSObject {
 
             // Create and perform sync if there is a self client.
             if let selfClientID = selfUserClient.remoteIdentifier {
-                setUpSyncAgent(clientID: selfClientID, asyncStreamEnabled: selfUserClient.asyncStreamCapable)
+                setUpSyncAgent(clientID: selfClientID)
             }
         }
     }
 
-    private func setUpSyncAgent(clientID: String, asyncStreamEnabled: Bool) {
+    private func setUpSyncAgent(clientID: String) {
         let clientSessionComponent = userSessionComponent.clientSessionComponent(
             clientID: clientID,
-            asyncStreamEnabled: asyncStreamEnabled,
             processorHandlers: .init(
                 onProcessedCallEvent: onProcessedCallEvent(callEventInfo:),
                 onSelfClientInvalidated: onSelfClientInvalidated,
                 onProcessedTypingUsers: onProcessedTypingUsers(typingUsersInfo:)
             ),
             onAuthenticationFailure: onAuthenticationFailure
         )
-
-        if asyncStreamEnabled {
-            // TODO: [WPB-17223] move this just after the migration is done
-            journal[.isSyncV3Enabled] = true
-        }
+        self.clientSessionComponent = clientSessionComponent
```

Comment:
this is a bit hacky, but maybe this can be improve with deterministic session loading @johnxnguyen